solidity/test/support/helpers.js

```diff
@@ -15,53 +27,39 @@ const ethjsUtils = require('ethereumjs-util')
-// (4) 0x0d1d4e623d10f9fba5db95830f7d3839406c6af2
-// (5) 0x2932b7a2355d6fecc4b5c0b6bd44cc31df247a2e
-// (6) 0x2191ef87e392377ec08e7c08eb105ef5448eced5
-// (7) 0x0f4f2ac550a1b4e2280d04c21cea7ebd822934b5
-// (8) 0x6330a553fc93768f612722bb8c2ec78ac90b3bbc
-// (9) 0x5aeda56215b167893e80b4fe645ba6d5bab767de
-
-// Private Keys
-// ==================
-// (0) c87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3
-// (1) ae6ae8e5ccbfb04590405997ee2d52d2b330726137b875053c36d94e974d162f
-// (2) 0dbbe8e4ae425a6d2687f1a7e3ba17bc98c673636790f1b8ad91193c05875ef1
-// (3) c88b703fb08cbea894b6aeff5a544fb92e78a18e19814cd85da83b71f772aa6c
-// (4) 388c684f0ba1ef5017716adb5d21a053ea8e90277d0868337519f97bede61418
-// (5) 659cbb0e2411a44db63778987b1e22153c086a95eb6b18bdf89de078917abc63
-// (6) 82d052c865f5763aad42add438569276c00d3d88a2d062d36b2bae914d58b8c8
-// (7) aa3680d5d48a8283413f7a108367c7299ca73f553735860a87b08f39395618b7
-// (8) 0f62d96d6675f32685bbdb8ac13cda7c23436f63efbb9d07700d8669ff12b7c4
-// (9) 8d5366123cb560bb606379f90a0bfd4769eecc0557f1b362dcae9012b548b1e5
-
-// HD Wallet
-// ==================
-// Mnemonic:      candy maple cake sugar pudding cream honey rich smooth crumble sweet treat
-// Base HD Path:  m/44'/60'/0'/0/{account_index}
-const accounts = eth.accounts
-
-export const defaultAccount = accounts[0]
-export const oracleNode = accounts[1]
-export const stranger = accounts[2]
-export const consumer = accounts[3]
+// https://github.com/ethereum/web3.js/issues/1119#issuecomment-394217563
+web3.providers.HttpProvider.prototype.sendAsync =
+  web3.providers.HttpProvider.prototype.send
+export const eth = web3.eth;
+
+const INVALIDVALUE = {
+  // If you got this value, you probably tried to use one of the variables below
+  // before they were initialized. Do any test initialization which requires
+  // them in a callback passed to Mocha's `before` or `beforeEach`.
+  // https://mochajs.org/#asynchronous-hooks
+  unitializedValueProbablyShouldUseVaribleInMochaBeforeCallback: null
+}
+export let [accounts, defaultAccount, oracleNode, stranger, consumer] =
+  Array(1000).fill(INVALIDVALUE)
+before(async function queryEthClientForConstants () {
+  accounts = (await eth.getAccounts());
+  [defaultAccount, oracleNode, stranger, consumer] = accounts.slice(0, 4)
+  })
+
 export const utils = Utils(web3.currentProvider)
 export const wallet = Wallet(PRIVATE_KEY, utils)
 export const deployer = Deployer(wallet, utils)
 
-export const bigNum = number => web3.toBigNumber(number)
+export const bigNum = number => web3.utils.toBN(number)
 
-export const toWei = number => bigNum(web3.toWei(number))
+// toWei(n) is n * 10**18, as a BN.
+export const toWei = number => bigNum(web3.utils.toWei(bigNum(number)))
 
-export const hexToInt = string => web3.toBigNumber(string)
+export const toUtf8 = web3.utils.toUtf8
+
+export const keccack = web3.utils.sha3
+
+export const hexToInt = string => bigNum(string).toNumber()
```

Comment:
Is `bigNum` not a `Number`? Sprinkling all these coercion functions show a lack of confidence in the code.

Reply:
No, it's a [`BN`](https://github.com/indutny/bn.js/#utilities). That seems to be the way web3 v1 seems to be going.